src/kilo.c

```diff
@@ -1009,13 +1010,60 @@ char *editorReadStringFromStatusBar(char *prefix) {
       break;
     }
 
+
+    case ARROW_UP: {
+      char *hist = getHistoryAt(history + 1);
+      if (hist == NULL) {
+        break;
+      }
+      history++;
+
+      // Copy the history string into the buffer.
+      int len = strlen(hist);
+      while (len + 1 > bufsz) {
+        str = realloc(str, bufsz *= 2);
+      }
+      memset(str, 0, endpos);
+      strcpy(str, hist);
+      endpos = len;
+      inspos = endpos;
+      break;
+    }
+
+    case ARROW_DOWN: {
+      if (history == 0) {
```

Comment:
Shouldn't we recover the original string here? Looks like this just zeros it out